src/components/App.js

```diff
@@ -1,104 +1,90 @@
-
-        const regExpHexShort = new RegExp(/^#[0-9A-F]{3}$/gi);
-        const regExpHexLong = new RegExp(/^#[0-9A-F]{6}$/gi);
-        // console.log('input ', [event.target.name],event.target.value, );
-        if (!(regExpHexShort.test(event.target.value) || regExpHexLong.test(event.target.value))) {
-            // if(!(regExpHexShort.test(event.target.value) || regExpHexLong.test(event.target.value))) {
-            //
-            // }
-            console.log('not valid', event.target.value)
-        } else {
-            console.log(' valid', event.target.value);
-            // this.setState ({
-            //     isDisabled: !this.state.isDisabled
-            // })
-        }
+  constructor() {
+    super();
+    this.state = {
+      inputValueFirst: "", //сюда должны получать значение из инпута
+      inputValueSecond: "", //сюда должны получать значение из инпута
+      isSubmitted: false,
+      isDisabled: false,
+      itemsList: [],
     };
+  }
 
-    // validate = () => {
-    //     let inputError = '';
-    // };
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
 
-    // deleteItem = item => {
-    //     const updateItemsList = this.state.itemsList.filter(function (item) {
-    //         return item.id !== movie.id;
-    //     });
-    //     console.log('updateItemsList', updateItemsList);
-    //     this.setState ({
-    //         itemsList: updateItemsList
-    //     });
-    // };
-    handleSubmit = (event) => {
-        event.preventDefault();
-        const arrayVar = this.state.itemsList;
-        arrayVar.push(<GradientItem inputValueFirst = {this.state.inputValueFirst}
-                                    inputValueSecond = {this.state.inputValueSecond}
-                                    deleteItem = {this.deleteItem}/>);
-
-        this.setState({
-            isSubmitted: true,
-            itemsList: arrayVar
-            // itemsArray: [...this.state.itemsArray, {GradientItem}]
-        })
+    const regExpHexShort = new RegExp(/^#[0-9A-F]{3}$/gi);
+    const regExpHexLong = new RegExp(/^#[0-9A-F]{6}$/gi);
+    if (
+      !(
+        regExpHexShort.test(event.target.value) ||
+        regExpHexLong.test(event.target.value)
+      )
+    ) {
+      console.log("not valid", event.target.value);
+    } else {
+      console.log(" valid", event.target.value);
+    }
+  };
 
-        //  const isValid = this.validate();
-        // if (isValid) {
-        //     console.log(this.state);
-        // }
-    };
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const arrayVar = this.state.itemsList;
+    arrayVar.push(
+      <GradientItem
+        inputValueFirst={this.state.inputValueFirst}
+        inputValueSecond={this.state.inputValueSecond}
+        deleteItem={this.deleteItem}
+      />
+    );
+    this.setState({
+      isSubmitted: true,
+      itemsList: arrayVar,
+    });
+  };
```

Comment:
Тут всё что тебе нужно, это добавить набор цветов в массив.
```suggestion
  handleSubmit = (event) => {
    event.preventDefault();
    
    const { primaryColor, secondaryColor, colorSets } = this.state;
    
    const colorSet = { 
      id: _uniqueId(), // добавь пакет lodash и импортируй из него эту функцию
      primary: primaryColor, 
      secondary: secondaryColor 
    };
    
    this.setState(prevState => ({
      colorSets: [...prevState.colorSets, colorSet],
      primaryColor: '',
      secondaryColor: '',
    ))
  };
```